pkg/storage/storage.go

```diff
@@ -143,3 +146,14 @@ func (s *Storage) GetSignedUrl(ctx context.Context, key string, expires time.Dur
 func (s *Storage) GetPublicUrl(key string) string {
 	return fmt.Sprintf(s.Config.URL_PREFIX, key)
 }
+
+func (s *Storage) GetFileKeyFromPublicUrl(imageURL string) (string, error) {
+	parsedURL, err := url.Parse(imageURL)
+	if err != nil {
+		return "", apperror.InternalServerError(err, "Failed to parse URL")
```

Comment:
return as normal error cuz it's external
```suggestion
return "", apperror.InternalServerError(err, "Failed to parse URL")